src/compiler/checker.ts

```diff
@@ -13120,7 +13120,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             if (baseConstructorIndexInfo) {
                 indexInfos = append(indexInfos, baseConstructorIndexInfo);
             }
-            if (symbol.flags & SymbolFlags.Enum && (getDeclaredTypeOfSymbol(symbol).flags & TypeFlags.Enum ||
+            if (symbol.flags & SymbolFlags.RegularEnum && (getDeclaredTypeOfSymbol(symbol).flags & TypeFlags.Enum ||

```

Comment:
I didn't even know that `preserveConstEnum` is a thing 😅 Looking at the example output presented in the docs:
```ts
"use strict";
var Album;
(function (Album) {
    Album[Album["JimmyEatWorldFutures"] = 1] = "JimmyEatWorldFutures";
    Album[Album["TubRingZooHypothesis"] = 2] = "TubRingZooHypothesis";
    Album[Album["DogFashionDiscoAdultery"] = 3] = "DogFashionDiscoAdultery";
})(Album || (Album = {}));
const selectedAlbum = 1 /* Album.JimmyEatWorldFutures */;
if (selectedAlbum === 1 /* Album.JimmyEatWorldFutures */) {
    console.log("That is a great choice.");
}
```

I think it would be OK to allow those index signatures with this flag. The call site wouldn't get substituted but there would be no `ReferenceError` thrown as the enum itself would still exist at runtime.

If that would be allowed then maybe this whole thing would deserve a specialized diagnostic when const enums declarations are elided from the source code? I'm not sure what's the process of adding a new diagnostic since those messages are internationalized etc.

Either way, I can make changes here but I'd prefer to do them when somebody chimes in with a confirmation that a specific change has to be done.

Reply:
To be clear, I'm not sure as to the legality either; this is reverse mapping (right?) and so I would think it's ambiguous what you get out. Really this is just a drive-by review.

> I'm not sure what's the process of adding a new diagnostic since those messages are internationalized etc.

You can just add them, and they'd get translated (eventually), though not recently.